libr/anal/fcn.c

```diff
@@ -1818,14 +1820,38 @@ R_API int r_anal_fcn_add_bb(RAnal *anal, RAnalFunction *fcn, ut64 addr, ut64 siz
 			r_anal_fcn_update_tinyrange_bbs (fcn);
 		}
 	}
-	if (!bb) {
-		bb = appendBasicBlock (anal, fcn, addr);
+	if (x86) {
+		if (bb) {
+			r_list_delete_data (fcn->bbs, bb);
+		}
+		ut8 *bbuf = malloc (size);
+		if (bbuf) {
```

Comment:
we should probably just return from the function if we can't even malloc something

Reply:
ok 54813aae942774a0a606310a6823ce19c95b036b